src/psyclone/domain/common/algorithm/psyir.py

```diff
@@ -36,18 +36,33 @@
 '''This module contains PSyclone Algorithm-layer-specific PSyIR classes.
 
 '''
-from psyclone.psyir.nodes import Call, Reference, DataNode
-from psyclone.psyir.symbols import TypeSymbol
+from psyclone.psyir.nodes import Call, Reference, DataNode, Literal, \
+    ArrayReference
+from psyclone.psyir.symbols import TypeSymbol, ContainerSymbol, \
+    GlobalInterface, RoutineSymbol
 from psyclone.errors import GenerationError
 
 
 class AlgorithmInvokeCall(Call):
-    '''An invoke call in a PSyclone Algorithm layer.'''
+    '''An invoke call in a PSyclone Algorithm layer.
 
+    :param routine: the routine that this call calls.
+    :type routine: py:class:`psyclone.psyir.symbols.RoutineSymbol`
+    :param parent: optional parent of this node in the PSyIR. Defaults \
+        to None.
+    :type parent: sub-class of :py:class:`psyclone.psyir.nodes.Node` \
+        or NoneType
+
+    '''
     _children_valid_format = "[KernelFunctor]*"
     _text_name = "AlgorithmInvokeCall"
     _colour = "green"
 
+    def __init__(self, routine, parent=None):
+        super(AlgorithmInvokeCall, self).__init__(routine, parent=parent)
+        self._routine_symbol = None
```

Comment:
I think these 2 attributes should probably not exist and have a index attribute instead. See previous comment.

Reply:
I agree that it makes sense to store the index separately. However, I think the routine and container symbols need to be stored somewhere - please see previous comment for context.

Reply:
I agree with this

Reply:
Updated.